finetune/full.py

```diff
@@ -27,6 +27,7 @@
 from scripts.prepare_alpaca import generate_prompt
 
 eval_interval = 600
+full_checkpointing = False
```

Comment:
I would opt for simplicity and store either the full thing or just the model. In general, we should treat full-finetuning as we do pretraining, i.e., implement the resuming functionality fully and correctly, including data handling. 

So I think we should decide here on whether we want to do full checkpointing (I vote for full), and in a follow up to this PR port over some of the important resuming logic from pretrain/tinyllama.py

Reply:
Ok, I reverted this part and keep the full checkpointing for now. But I'm preparing a follow up now where I make the full.py as close as possible to our pretraining script (since it is very similar)